src/apps/js_generic/js_generic.cpp

```diff
@@ -82,9 +82,26 @@ namespace ccfapp
     JS_FreeValue(ctx, exception_val);
   }
 
-  struct JSAutoFree
+  struct JSAutoFreeRuntime
   {
-    JSAutoFree(JSContext* ctx) : ctx(ctx) {}
+    JSRuntime* rt;
+
+    JSAutoFreeRuntime(JSRuntime* rt) : rt(rt) {}
```

Comment:
can you do the same things with a std::unique_ptr that specifics a custom deleter?

Reply:
You can, but it's not super satisfying because you pay for an extra pointer on every object, and you can't use make_unique, so it's both a little awkward to use and a somewhat expensive.